Classes/Service/VarnishBanService.php

```diff
@@ -118,14 +117,7 @@ public function banByTags(array $tags, $domains = null)
         }
 
         // Set specific domain before invalidating tags
-        if ($domains) {
-            $this->varnishProxyClient->setDefaultBanHeader(ProxyClient\Varnish::HTTP_HEADER_HOST, is_array($domains) ? '^(' . implode('|', $domains) . ')$' : $domains);
-        }
-        $this->tagHandler->invalidateTags($tags);
-        // Unset specific domain after invalidating tags
-        if ($domains) {
-            $this->varnishProxyClient->setDefaultBanHeader(ProxyClient\Varnish::HTTP_HEADER_HOST, ProxyClient\Varnish::REGEX_MATCH_ALL);
-        }
+        $this->cacheInvalidator->invalidate($this->getHostHeader($domains))->invalidateTags($tags);
```

Comment:
So yeah, can confirm.

`invalidate()` will queue a request that bans everything matching the X-Host header.

The way I see it there's no option to set the X-Host header when banning by tags. So we could either:
* extend the FOS ProxyClient\Varnish to allow that
* manually generate the X-Cache-Tags header and use `invalidate()` instead of `invalidateByTags()`, although then we have to also take care of escaping and splitting the request into multiple ones when the header size is too large

Reply:
Hey,

I just had time to look into this and it seems we need to implement it ourselfs as the fos/http-cache people don't think this belongs into their library. FriendsOfSymfony/FOSHttpCache#476

Reply:
or you use the domain name in the tags. the performance optimized xkey tag system of varnish does not allow to restrict which tags to invalidate, so supporting it on php side would be difficult.

Reply:
Hi @dbu,
I already ran into problems in some customer projects because pages had a huge number of tags (tends to happen with pages that list content from other pages). When requesting the page, the response header size can easily exceed the allowed header size of the proxy. Therefore I don't think adding the whole domain to the tags is a good idea. 

@flyinggecko Maybe we could assign a number to the Sites in Neos and add those to the tags? That would only increase the length a tiny bit.

Reply:
As I understand this is not really easy as neos allows multiple domains per site and then the tags are the same (as all pages/nodes are the same), but the domain is different. But, yeah, this is something the actual maintainers have to think about. ;)

Reply:
But even when using multiple domains per site, the tags of the rendered content across those domains should be the same? So in my opinion this could work. 

But I agree, @aertmann what do you think?

Reply:
@nlx-lars We could try that, but I would need a bit of guidance there, as I'm relatively new to Neos.

Reply:
if you have something shorter but unique, that would be better. if there are site ids or somesuch, it would allow you to even support the multiple domains per site, which would not work if you ban tags with domain restriction (or further complicate the thing because its a list of domains).

RE long tag header tags: newer versions of FOSHttpCache have a config for the max header length and split the tags header into multiple headers if necessary. you could do a hashing on the tags if individual tags are very long. as long as the hashing happens both when tagging and when invalidating, the result is the same (and hash collisions at most lead to a small performance loss, never more damage). if you have tons of individual tags, you want for sure use xkey to not bog down varnish with ban requests - and then the only way is to include the side id (or domain) in the tags.

Reply:
Hey, it took a bit but I have some time to spend on this again.

Did you have a preferred resolution for this? I would propose to use something git-like (shortened sha1-hash). It is a bit more overhead for the generation of the tags, but I think a lot easier then some incrementing id that we need to keep track of. Or another way could be using the last section of the persistent_object_identifier which might also be enough.

Reply:
Hey @flyinggecko,

in my opinion, something along the lines of `substr(sha1($siteIdentifier . $cacheTag), $configurableCacheTagLength)` should be enough. Where `$siteIdentifier` is either the `persistence_object_identifier` or `siteNodeName` of a neos site.

What do you think, @aertmann?